pkg/controller/nodelifecycle/config/v1alpha1/defaults.go

```diff
@@ -38,7 +38,7 @@ func RecommendedDefaultNodeLifecycleControllerConfiguration(obj *kubectrlmgrconf
 		obj.PodEvictionTimeout = metav1.Duration{Duration: 5 * time.Minute}
 	}
 	if obj.NodeMonitorGracePeriod == zero {
-		obj.NodeMonitorGracePeriod = metav1.Duration{Duration: 40 * time.Second}
+		obj.NodeMonitorGracePeriod = metav1.Duration{Duration: 50 * time.Second}
```

Comment:
The linked issue does a great job explaining *why* the current default is not safe, but the docs here or on the flag don't help users at all. This is now a parameter which can be tuned which has some settings we *know* are dangerous that we're not detecting / warning about at all.

At the very least, we need comments here and in the flag help explaining that this should be greater than HTTP2_PING_TIMEOUT_SECONDS (default 30) + HTTP2_READ_IDLE_TIMEOUT_SECONDS (default 15), and why.

Those values are *also* tunable... at startup, should we check that the effective values for the HTTP2 timeouts and this grace period are coherent and warn if they are not?

Reply:
+1

Reply:
> The linked issue does a great job explaining why the current default is not safe, but the docs here or on the flag don't help users at all. This is now a parameter which can be tuned which has some settings we know are dangerous that we're not detecting / warning about at all.

I have added a few comments and updated the flag description as well

> Those values are also tunable... at startup, should we check that the effective values for the HTTP2 timeouts and this grace period are coherent and warn if they are not?
@dchen1107 @liggitt Would it make sense to have this in a separate PR, or just update this one?

Reply:
Since I updated the flag description, I believe the schema description in zz_generated.openapi.go should have been updated when I ran `make update`. Is this the correct way or did I do something wrong?

Reply:
Hello @dchen1107 @liggitt 
I updated the PR including why the change is required correctly.

Reply:
LGTM and thanks!